src/server.c

```diff
@@ -1672,8 +1672,27 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
         if (listLength(server.clients_waiting_acks) && prev_fsynced_reploff != server.fsynced_reploff) dont_sleep = 1;
     }
 
-    /* Handle writes with pending output buffers. */
-    handleClientsWithPendingWritesUsingThreads();
+    /* Handle writes with pending output buffers.
+     * In the first phase, we will try to leverage io_uring to handle fsync asynchronously,
+     * just skip the client that has pending writes. */
+    handleClientsWithPendingWritesUsingThreads(1);
+
+    if (server.aof_state == AOF_ON && server.aof_fsync == AOF_FSYNC_ALWAYS && server.io_uring_enabled) {
+        /* Wait io_uring_prep_fsync finished. */
+        int ret = ioUringWaitFsyncBarrier(server.io_uring);
```

Comment:
I couldn't find my previous comment so posting it here again: I think ioUringWaitFsyncBarrier can get stalled so we need some metrics to understand its impact on the main thread. at the minimum, I think we should implement similar latency doctor logic but maybe it is worthwhile to have a single normal `INFO` metric that records the time the main thread spends in fsync (io_uring or not)?

Reply:
@PingXie I am thinking how to record the time the main thread spent on io_uring fsync. The process is like below:
1. main thread submit fsync request to io_uring.
2. kernel offload the fsync to DMA.
3. main thread continue to handleClientsWithPendingWritesUsingThreads.
4. main thread wait for io_uring fysnc to finish. 

We only record the duration in step 4, line 1681 ~ 1683?